### PR TITLE
Fix @osx and @windows dep warn

### PR DESCRIPTION
**Pull request by ranjanan**
`@osx_only` and `@windows` to `@compat @static if is_apple` and `@compat @static is_windows`

### CODE REVIEW

File: deps/build.jl [hunk 1]
```diff
@@ -10,18 +10,20 @@ provides(AptGet, Dict("libgsl0ldbl"=>libgsl, "libgsl0-dev" =>libgsl, "gsl-bin"=>
 provides(Yum, "gsl-devel", libgsl)
 provides(Pacman, "gsl", libgsl)
 
-@osx_only begin
+@compat @static if is_apple() begin
```
Comment (tkelman) [inline deps/build.jl]: need to add Compat, with minimum version, to REQUIRE. don't need either compat or static macro here, but do need using Compat

File: deps/build.jl [hunk 2]
```diff
@@ -10,18 +10,20 @@ provides(AptGet, Dict("libgsl0ldbl"=>libgsl, "libgsl0-dev" =>libgsl, "gsl-bin"=>
 provides(Yum, "gsl-devel", libgsl)
 provides(Pacman, "gsl", libgsl)
 
-@osx_only begin
+if is_apple() begin
```
Comment (tkelman) [inline deps/build.jl]: the begin is redundant now

**Comment by yuyichao** [comment]
LGTM. Ping @jiahao

**Comment by jiahao** [comment]
Thanks!

**Comment by ranjanan** [comment]
Thanks. Could you please tag a new release?

**Comment by ranjanan** [comment]
Bump @jiahao, could you tag a new release please?

**Pull request merged**